File: Hackerrank Problem Solving/8.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'miniMaxSum' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#

def miniMaxSum(lst):
    # Write your code here
    x = sum(lst)
    print (x-(max(lst)), x-(min(lst)))

if __name__ == '__main__':

    arr = list(map(int, input().rstrip().split()))

    miniMaxSum(arr)

File: C++/Basic Problems in C++/8.cpp
int trap(vector<int>& nums) {
        vector<int>ml(nums.size()),mr(nums.size());
        ml[0]=nums[0];
        
        for(int i=1;i<nums.size();i++)
            ml[i]=max(ml[i-1],nums[i]);
        
        
        mr[nums.size()-1]=nums[nums.size()-1];
        
        for(int i=nums.size()-2;i>=0;i--)
            mr[i]=max(nums[i],mr[i+1]);
        
        int water=0;
        
        for(int i=1;i<nums.size()-1;i++){
            water+=min(mr[i],ml[i])-nums[i];
        }
        return water;
    }